### PR TITLE
Improve user

### PR DESCRIPTION
**Pull request by rustamwin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️/❌
| New feature?  | ✔️/❌
| Breaks BC?    | ✔️/❌
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

### CODE REVIEW

File: src/User.php [hunk 1]
```diff
@@ -267,7 +267,7 @@ private function afterLogout(IdentityInterface $identity): void
      * @param IdentityInterface $identity the identity information to be associated with the current user.
      * In order to indicate that the user is guest, use {{@see GuestIdentity}}.
      */
-    public function switchIdentity(IdentityInterface $identity): void
+    private function switchIdentity(IdentityInterface $identity): void
```
Comment (roxblnfk) [inline src/User.php:270]: This method should be public

File: src/User.php [hunk 2]
```diff
@@ -346,4 +348,18 @@ public function can(string $permissionName, array $params = []): bool
 
         return $this->accessChecker->userHasPermission($this->getId(), $permissionName, $params);
     }
+
+    public function authTimeout(int $timeout = null): self
```
Comment (roxblnfk) [inline src/User.php:352]: Is it setter? Why not `setAuthTimeout()`?

**Review by roxblnfk — CHANGES_REQUESTED** [review]

File: src/User.php (hunk 2 above)
Comment (vjik) [reply src/User.php:352]: Also for method `absoluteAuthTimeout`.

**Comment by vjik** [comment]
Maybe remove setters for `authTimeout`/`absoluteAuthTimeout`, move him to constructor and set via app configuration?

**Comment by rustamwin** [comment]
> Maybe remove setters for authTimeout/absoluteAuthTimeout, move him to constructor and set via app configuration?

They are optional so it doesn't make sense to move to constructor.

**Comment by vjik** [comment]
> > Maybe remove setters for authTimeout/absoluteAuthTimeout, move him to constructor and set via app configuration?
> 
> They are optional so it doesn't make sense to move to constructor.

In constructor with default values

File: src/User.php (hunk 1 above)
Comment (rustamwin) [reply src/User.php]: Decided to leave it as it is.

**Review by samdark — APPROVED** [review]

**Pull request merged**